lib/sqlalchemy/sql/sqltypes.py

```diff
@@ -155,15 +162,19 @@ class Indexable(TypeEngineMixin):
     class Comparator(TypeEngine.Comparator[_T]):
         __slots__ = ()
 
-        def _setup_getitem(self, index):
+        def _setup_getitem(self, index: _ColumnExpressionArgument[_T]) -> None:
             raise NotImplementedError()
 
-        def __getitem__(self, index):
+        def __getitem__(
+            self, index: _ColumnExpressionArgument[_T]
+        ) -> ColumnElement[_T]:
             (
                 adjusted_op,
                 adjusted_right_expr,
                 result_type,
-            ) = self._setup_getitem(index)
+            ) = self._setup_getitem(
+                index
+            )  # type: ignore
```

Comment:
**mike bayer** (zzzeek) wrote:

fix return type for _setup_getitem, shouldnt need a type ignore here

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4509